pypdf/_doc_common.py

```diff
@@ -497,7 +497,13 @@ def _get_named_destinations(
             while i < len(names):
                 key = cast(str, names[i].get_object())
                 i += 1
-                if not isinstance(key, str):
+                original_key = key
+                if isinstance(key, bytes):
```

Comment:
Is the key actually `bytes` or our specific subclass `pypdf.generic._base.ByteStringObject`? If it is a `ByteStringObject`, doing `str(key)` should succeed in every case.